api/store/mongo/store.go

```diff
@@ -1340,9 +1358,43 @@ func (s *Store) CreateNamespace(ctx context.Context, namespace *models.Namespace
 	return namespace, err
 }
 func (s *Store) DeleteNamespace(ctx context.Context, namespace string) error {
```

Comment:
You can use `DeleteMany` to removes all documents that match the filter from a collection instead of iterating through a cursor for a performance reasons